shared/src/business/useCases/practitioners/deletePractitionerDocumentInteractor.test.ts

```diff
@@ -0,0 +1,56 @@
+import { ROLES } from '../../entities/EntityConstants';
+import { UnauthorizedError } from '../../../errors/errors';
+import { applicationContext } from '../../test/createTestApplicationContext';
+import { deletePractitionerDocumentInteractor } from './deletePractitionerDocumentInteractor';
+
+describe('deletePractitionerDocumentInteractor', () => {
+  const testUser = {
+    role: ROLES.admissionsClerk,
+    userId: 'admissionsclerk',
+  };
+  const practitionerDocumentFileId = '3da3e303-5da9-4b8b-99a5-d4b65c449619';
+  const barNumber = 'PT76543';
+
+  beforeAll(() => {
+    applicationContext.getCurrentUser.mockReturnValue(testUser);
+  });
+
+  it('should throw an unauthorized error when the user does not have permission to update practitioner documents', async () => {
+    const testPetitionerUser = {
+      role: ROLES.petitioner,
+      userId: 'petitioner',
+    };
+    applicationContext.getCurrentUser.mockReturnValueOnce(testPetitionerUser);
+
+    await expect(
+      deletePractitionerDocumentInteractor(applicationContext, {
+        barNumber,
+        practitionerDocumentFileId,
+      }),
+    ).rejects.toThrow(UnauthorizedError);
+  });
+
+  it('should delete the practitioner document from s3', async () => {
+    await deletePractitionerDocumentInteractor(applicationContext, {
+      barNumber,
+      practitionerDocumentFileId,
+    });
+    expect(
+      applicationContext.getPersistenceGateway().deleteDocumentFromS3.mock
+        .calls[0][0],
+    ).toMatchObject({
+      key: practitionerDocumentFileId,
+    });
+  });
```

Comment:
It seems like a deviation from the Clean Architecture approach that has gone into this project to mention S3 explicitly. In theory we could use another file store and not have to change this function's name, right? Is `deleteDocumentFile` more fitting?

Reply:
Renamed to `deleteDocumentFile`